backends/tofino/bf-asm/tables.h

```diff
@@ -1001,7 +1001,8 @@ class Table {
     virtual int get_tcam_id() const { BUG("%s not a TCAM table", name()); }
 
     const std::vector<const p4_param *> find_p4_params(std::string s, std::string t = "",
-                                                       int start_bit = -1, int width = -1) const {
+                                                       unsigned start_bit = -1,
```

Comment:
This seems wrong, as the default value of `-1` is not unsigned.  With this the `start_bit <= -1` test below will always be false.